jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/operation/GenerateWebAppRunSteps.java

```diff
@@ -0,0 +1,27 @@
+package com.cerner.jwala.ui.selenium.steps.operation;
+
+import com.cerner.jwala.ui.selenium.component.JwalaUi;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.openqa.selenium.By;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Created by Jedd Cuison on 8/14/2017
+ */
+public class GenerateWebAppRunSteps {
+
+    @Autowired
+    private JwalaUi jwalaUi;
+
+    @When("^I click the generate web application button of \"(.*)\"$")
+    public void clickGenerateWebAppBtnOfGroup(final String webAppName) {
+        jwalaUi.click(By.xpath("//tr/td[text()='" + webAppName + "']/following-sibling::td[3]/div/button[@title='Generate and deploy the webapp resources.']"));
```

Comment:
We have to be concise. I usually check if the gears are "called" before using click when ready since click when ready has "extra checking" that we we might not need in certain scenarios.

In short, we shouldn't use clickWhenReady  based on assumption. Check first if it really gets called. You have to check yours.